actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -406,6 +407,16 @@ def ssl?
       super || scheme == 'wss'.freeze
     end
 
+    def method_missing(method, *args)
+      super unless Rack::Request.public_instance_methods.include?(method)
+      ActiveSupport::Deprecation.warn(<<-MESSAGE.strip_heredoc)
```

Comment:
Note the control flow. Now we're calling super but not returning, so we'll invoke the superclass method_missing then emit a deprecation warning.